instr.ml

```diff
@@ -322,3 +322,82 @@ let checkpoint_label pc =
 
 let independent instr exp =
   VarSet.is_empty (VarSet.inter (changed_vars instr) (expr_vars exp))
+
+class mapper = object (m)
+  method variable x = x
```

Comment:
Do you think that I could separate three/ways into binding, use, and assignment positions? If I formulated the mut-to-const optimization above as transforming uses but not assignments, would that be good with you?